src/pkg/packager/actions.go

```diff
@@ -181,28 +190,42 @@ func actionGetCfg(cfg types.ZarfComponentActionDefaults, a types.ZarfComponentAc
 	return cfg
 }
 
-func actionRun(ctx context.Context, cfg types.ZarfComponentActionDefaults, cmd string) (string, error) {
+func actionRun(ctx context.Context, cfg types.ZarfComponentActionDefaults, cmd string, spinner *message.Spinner) (string, error) {
 	var shell string
 	var shellArgs string
 
 	if runtime.GOOS == "windows" {
 		shell = "powershell"
 		shellArgs = "-Command"
+		message.Debug("Running command in PowerShell: %s", cmd)
 	} else {
 		shell = "sh"
 		shellArgs = "-c"
+		message.Debug("Running command in shell: %s", cmd)
 	}
 
 	execCfg := exec.Config{
-		Print: !cfg.Mute,
-		Env:   cfg.Env,
-		Dir:   cfg.Dir,
+		Env: cfg.Env,
+		Dir: cfg.Dir,
 	}
-	output, errOut, err := exec.CmdWithContext(ctx, execCfg, shell, shellArgs, cmd)
-	// Dump the command output in debug if output not already streamed.
-	if cfg.Mute {
-		message.Debug(output, errOut)
+
+	if !cfg.Mute {
+		execCfg.Stdout = spinner
+		execCfg.Stderr = spinner
 	}
 
-	return output, err
+	out, errOut, err := exec.CmdWithContext(ctx, execCfg, shell, shellArgs, cmd)
+	// Dump final complete output.
+	message.Debug(cmd, out, errOut)
+
+	return out, err
+}
+
+func escapeCmdForPrint(cmd string) string {
+	cmdEscaped := strings.ReplaceAll(cmd, "\n", "; ")
+	// Truncate the command if it is longer than 60 characters (to fit well in 80 chars)
```

Comment:
`    Waiting for command "" (timeout: 000s) (000s)` is 49 chars long so would recommend maxing this out at ~30 chars instead for 80 columns (or shortening that printout).

Reply:
Yeah I actually played with that for a bit last night too, it looked super dumb to truncate scripts that early and adding conditional logic for the different lengths seemed sort of silly. We have lots of text that runs over 80 right now

Reply:
Then the comment should say 110 chars not 80 ;)